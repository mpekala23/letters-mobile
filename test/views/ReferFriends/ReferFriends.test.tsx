import React from 'react';
import { render, fireEvent, toJSON } from '@testing-library/react-native';
import { ReferFriendsScreen } from '@views';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { facebookShare } from '@api';

const mockStore = configureStore([]);

jest.mock('@api', () => ({
  facebookShare: jest.fn(),
}));

const setup = (contactOverrides = []) => {
  const navigation = { navigate: jest.fn() };
  const contact = {
    firstName: 'First',
    lastName: 'Last',
    inmateNumber: '6',
    relationship: 'Brother',
    ...contactOverrides,
  };

  const initialContactState = {
    active: contact,
  };

  const store = mockStore({
    contact: initialContactState,
  });

  const StoreProvider = ({ children }: { children: JSX.Element }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return {
    navigation,
    store,
    ...render(
      <ReferFriendsScreen
        navigation={navigation}
        contactState={initialContactState}
      />,
      {
        wrapper: StoreProvider,
      }
    ),
  };
};

describe('ReferFriends screen', () => {
  // it('should match snapshot', () => {
  //   const { container } = setup();
  //   const tree = toJSON(container);
  //   expect(tree).toMatchSnapshot();
  // });

  it('should return to SingleContact screen when done is pressed', async () => {
    const { navigation, getByText } = setup();
    const doneButton = getByText('Done');
    fireEvent.press(doneButton);
    expect(navigation.navigate).toHaveBeenCalledWith('SingleContact');
  });

  it('should make api call on share press', async () => {
    const { getByText } = setup();
    const shareUrl =
      'https://www.facebook.com/sharer/sharer.php?u=letters.ameelio.org';
    const shareButton = getByText('Share on Facebook');
    fireEvent.press(shareButton);
    expect(facebookShare).toHaveBeenCalledWith(shareUrl);
  });
});
