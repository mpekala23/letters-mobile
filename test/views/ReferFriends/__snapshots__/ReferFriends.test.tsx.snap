// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferFriends screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 16,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-around",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={171}
        bbWidth={218}
        fill="none"
        focusable={false}
        height={171}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "left": 52,
              "position": "absolute",
              "top": -5,
              "zIndex": 999,
            },
            Object {
              "flex": 0,
              "height": 171,
              "width": 218,
            },
          ]
        }
        vbHeight={171}
        vbWidth={218}
        width={218}
        xml="<svg width=\\"218\\" height=\\"171\\" viewBox=\\"0 0 218 171\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
<path d=\\"M61.0875 92C38.0946 92 -15.7744 135.961 7.87537 141.67C32.5738 147.633 58.9135 140.338 67 139.755\\" stroke=\\"#6D89B1\\" stroke-width=\\"3\\" stroke-dasharray=\\"4 4\\"/>
<path d=\\"M113.675 114L79.3421 135.609L58 124.084L113.675 114Z\\" fill=\\"#6D89B1\\"/>
<path d=\\"M113.675 116.521L80.27 138.49L77.9502 160.82L113.675 116.521Z\\" fill=\\"#6D89B1\\"/>
</svg>"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M61.0875 92C38.0946 92 -15.7744 135.961 7.87537 141.67C32.5738 147.633 58.9135 140.338 67 139.755"
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={4285368753}
            strokeDasharray={
              Array [
                "4",
                "4",
              ]
            }
            strokeWidth={3}
          />
          <RNSVGPath
            d="M113.675 114L79.3421 135.609L58 124.084L113.675 114Z"
            fill={4285368753}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGPath
            d="M113.675 116.521L80.27 138.49L77.9502 160.82L113.675 116.521Z"
            fill={4285368753}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        pointerEvents="auto"
      >
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F4F4F4",
              "borderColor": "white",
              "borderRadius": 65,
              "borderWidth": 6,
              "height": 130,
              "justifyContent": "center",
              "overflow": "hidden",
              "width": 130,
            }
          }
          testID="profilePicture"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins-Regular",
                },
                Object {
                  "color": "white",
                  "fontSize": 36,
                  "fontWeight": "500",
                },
              ]
            }
          >
            FL
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flex": 0,
            "marginBottom": 100,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Bold",
              },
              Object {
                "fontSize": 23,
                "textAlign": "center",
              },
            ]
          }
        >
          Your letter is on the way
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Regular",
              },
              Object {
                "color": "#6D6D6D",
                "fontSize": 14,
                "marginTop": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Your letter is estimated to arrive on
           
          Jul 12
          . 
          Thank you for trusting Ameelio.
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "width": "100%",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.7}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF7171",
                "borderRadius": 16,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 5,
                "paddingHorizontal": 14,
              },
              Object {},
              Object {
                "width": "100%",
              },
              Object {},
              Object {
                "elevation": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.22,
              },
            ]
          }
          testID="clickable"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins-Medium",
                },
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "padding": 6,
                },
                Object {},
                Object {},
                Object {},
              ]
            }
          >
            Share on Facebook
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.7}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#FF7171",
                "borderRadius": 16,
                "borderWidth": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 5,
              },
              Object {},
              Object {
                "width": "100%",
              },
              Object {},
              Object {
                "elevation": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 0,
                },
                "shadowOpacity": 0.22,
              },
            ]
          }
          testID="clickable"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins-Medium",
                },
                Object {
                  "color": "#FF7171",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "padding": 6,
                },
                Object {},
                Object {},
                Object {},
              ]
            }
          >
            Done
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
